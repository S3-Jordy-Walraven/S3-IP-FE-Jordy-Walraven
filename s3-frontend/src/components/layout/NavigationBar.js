import React from "react";
import { Container, Nav, Navbar, NavDropdown } from "react-bootstrap";
import { Link } from "react-router-dom";
import { userContext } from "../../context/userContext";

const NavigationBar = (props) => {
  const User = React.useContext(userContext);

  function logout() {
    User.userLogout();
  }

  return (
    <div>
      <Navbar
        collapseOnSelect
        expand="lg"
        className="general-dark-bg"
        variant="dark"
      >
        <Container>
          <Navbar.Brand as={Link} to="/">
            Home
          </Navbar.Brand>
          <Nav.Item
            className="text-light nav-link"
            as={Link}
            to="effect/upload"
          >
            Upload
          </Nav.Item>
          <Navbar.Toggle aria-controls="responsive-navbar-nav" />
          <Navbar.Collapse id="responsive-navbar-nav">
            <Nav className="me-auto"></Nav>
            <Nav>
              <userContext.Consumer>
                {({ user }) => {
                  return user == null ? (
                    <Nav.Item className="text-light">
                      Je bent uitgelogd
                    </Nav.Item>
                  ) : (
                    <p className="text-light">
                      Je bent ingelogd als {user.given_name}
                    </p>
                  );
                }}
              </userContext.Consumer>
              <NavDropdown title="Profile" id="collasible-nav-dropdown">
                <userContext.Consumer>
                  {({ user }) => {
                    return user == null ? (
                      <NavDropdown.Item as={Link} to="sign-up">
                        Login
                      </NavDropdown.Item>
                    ) : (
                      <NavDropdown.Item>
                        <button className="border-0" onClick={logout}>
                          Logout
                        </button>
                      </NavDropdown.Item>
                    );
                  }}
                </userContext.Consumer>
              </NavDropdown>
            </Nav>
          </Navbar.Collapse>
        </Container>
      </Navbar>
    </div>
  );
};

export default NavigationBar;
