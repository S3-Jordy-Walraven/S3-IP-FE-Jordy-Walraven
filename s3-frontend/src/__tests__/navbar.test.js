import { render, screen, cleanup } from "@testing-library/react";
import renderer from "react-test-renderer";
import NavigationBar from "../components/layout/NavigationBar";
import { BrowserRouter as Router } from "react-router-dom";

afterEach(cleanup);

test("should render app", () => {
    render(<Router><NavigationBar /></Router>);
    const canvasElement = screen.getByTestId("navBar");
    expect(canvasElement).toBeInTheDocument();
    expect(canvasElement).toHaveTextContent("Home");
    expect(canvasElement).toHaveTextContent("Upload");
});

test("matches snapshot", () => {
    const tree = renderer.create(<Router><NavigationBar /></Router>).toJSON();
    expect(tree).toMatchSnapshot();
});
