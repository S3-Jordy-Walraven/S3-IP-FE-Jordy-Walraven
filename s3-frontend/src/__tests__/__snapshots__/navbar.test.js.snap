// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  data-testid="navBar"
>
  <nav
    className="navbar navbar-expand-lg navbar-dark"
    style={
      Object {
        "backgroundColor": "#212D3A",
        "paddingBottom": "0.5%",
        "paddingTop": "0.5%",
      }
    }
  >
    <div
      className="container"
    >
      <a
        className="navbar-brand"
        href="/"
        onClick={[Function]}
      >
        Home
      </a>
      <button
        aria-controls="responsive-navbar-nav"
        aria-label="Toggle navigation"
        className="navbar-toggler collapsed"
        onClick={[Function]}
        type="button"
      >
        <span
          className="navbar-toggler-icon"
        />
      </button>
      <div
        aria-expanded={null}
        className="navbar-collapse collapse"
        id="responsive-navbar-nav"
      >
        <div
          className="me-auto navbar-nav"
          onKeyDown={[Function]}
        />
        <div
          className="navbar-nav"
          onKeyDown={[Function]}
          style={
            Object {
              "color": "white",
            }
          }
        >
          <a
            className="text-light nav-link nav-item"
            href="/effect/upload"
            onClick={[Function]}
            style={
              Object {
                "marginRight": "5%",
              }
            }
          >
            Upload
          </a>
          <div
            className="MuiAvatar-root MuiAvatar-circular MuiAvatar-colorDefault css-1r7xqj7-MuiAvatar-root"
          >
            <svg
              aria-hidden={true}
              className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiAvatar-fallback css-10mi8st-MuiSvgIcon-root-MuiAvatar-fallback"
              data-testid="PersonIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M12 12c2.21 0 4-1.79 4-4s-1.79-4-4-4-4 1.79-4 4 1.79 4 4 4zm0 2c-2.67 0-8 1.34-8 4v2h16v-2c0-2.66-5.33-4-8-4z"
              />
            </svg>
          </div>
          <div
            bg="#212D3A"
            className="nav-item dropdown"
          >
            <a
              aria-expanded={false}
              className="dropdown-toggle nav-link"
              data-rr-ui-event-key={null}
              href="#"
              id="navbarScrollingDropdown"
              onClick={[Function]}
              onKeyDown={[Function]}
              role="button"
              tabIndex={0}
            />
          </div>
        </div>
      </div>
    </div>
  </nav>
</div>
`;
